src/Libraries/RevitNodesUI/GenericClasses.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Autodesk.DesignScript.Geometry;
+using Autodesk.DesignScript.Runtime;
+using DSRevitNodesUI;
+using RVT = Autodesk.Revit.DB;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+
+using Dynamo.Utilities;
+using Dynamo.Models;
+using ProtoCore.AST.AssociativeAST;
+
+namespace DSRevitNodesUI
+{
+
+    /// <summary>
+    /// Generic UI Dropdown node baseclass for Revit Elements.
+    /// This class populates a dropdown with all Revit elements of the specified type.
+    /// It uses a filtered element collector filtering by class.
+    /// </summary>
+    public abstract class CustomRevitElementDropDown : RevitDropDownBase
+    {
+        /// <summary>
+        /// Generic Revit Element Class Dropdown Node
+        /// </summary>
+        /// <param name="name">Name of the Node</param>
+        /// <param name="elementType">Type of Revit Element to display</param>
+        public CustomRevitElementDropDown(string name, Type elementType) : base(name) { this.ElementType = elementType; PopulateItems(); }
+
+        /// <summary>
+        /// Constant message for missing Types
+        /// </summary>
+        private const string noTypes = "No Types available.";
```

Comment:
this should be moved to the .resx files in properties/resources for localization.